airbyte-server/src/main/java/io/airbyte/server/apis/factories/ConnectionApiFactory.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright (c) 2022 Airbyte, Inc., all rights reserved.
+ */
+
+package io.airbyte.server.apis.factories;
+
+import io.airbyte.analytics.TrackingClient;
+import io.airbyte.commons.version.AirbyteVersion;
+import io.airbyte.config.Configs.WorkerEnvironment;
+import io.airbyte.config.helpers.LogConfigs;
+import io.airbyte.config.persistence.ConfigRepository;
+import io.airbyte.config.persistence.SecretsRepositoryReader;
+import io.airbyte.config.persistence.SecretsRepositoryWriter;
+import io.airbyte.config.persistence.StatePersistence;
+import io.airbyte.persistence.job.JobPersistence;
+import io.airbyte.server.apis.ConnectionApiImpl;
+import io.airbyte.server.scheduler.EventRunner;
+import io.airbyte.server.scheduler.SynchronousSchedulerClient;
+import java.util.Map;
+import org.glassfish.hk2.api.Factory;
+import org.slf4j.MDC;
+
+public class ConnectionApiFactory implements Factory<ConnectionApiImpl> {
```

Comment:
Is this going to be replaced with a [Micronaut Factory](https://docs.micronaut.io/latest/api/io/micronaut/context/annotation/Factory.html) when this is all said-and-done?